src/cript/api/api.py

```diff
@@ -1172,35 +1159,40 @@ def delete(self, node) -> None:
         self.delete_node_by_uuid(node_type=node.node_type_snake_case, node_uuid=str(node.uuid))
 
     @beartype
-    def delete_node_by_uuid(self, node_type: str, node_uuid: str) -> None:
+    def delete_node_by_uuid(self, node_type: Union[Any, str], node_uuid: str) -> None:
```

Comment:
ideally, you don't have to specify the node type at all.
Check out the class attribute `UUIDBaseNode._uuid_cache`. 
It is a dictionary that maps UUID to all nodes that have been encountered with the current SDK session.

Use that to look up the node by uuid.

Only case this doesn't work, would be if the node is only present in the backend, in that case a simple `GET` to the back end should do the trick.

Reply:
that I did not think about. I'm not sure if that should go into this method or the method that is calling it. We could also have a utility method that uses this method as a wrapper that both looks up the node by uuid and then deletes it.

I really like this delete because I think that is really useful, and this method is really clean and takes only what is needed and deletes it.

if we want to have delete method that both looks up the UUID and then performs a delete, then I'd be in favor of making another method that does that, but would prefer if the user passes in their arguments instead so it is no room for ambiguity.

Reply:
The reason I am not a big fan of this delete is, that a) it introduces code duplication with the snake case and b) it puts burden on the user (providing the node type).
And we can eliminate this burden and should do that, because i) it is easier on the user and ii) if we don't the user can for example give you a wrong node type and debugging this is annoying for a user.

Reply:
I hear that, I'm kind of thinking that since I already wrote it, it is clean, and could be useful to keep it and provide a new method called `find_and_delete_node_by_uuid` that finds the node by uuid and deletes it. 

I'm not sure if it'll create code duplication with multiple delete methods or give more options. I have a feeling that this method will end up being inevitable at the end because other methods will just be a wrapper around this, but honestly not sure. What are your thoughts?